dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/HttpUtils.java

```diff
@@ -18,41 +18,116 @@
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
 import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.DefaultHttpClient;
 import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
 import org.apache.http.util.EntityUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
 import java.io.IOException;
-
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
 /**
  * http utils
  */
 public class HttpUtils {
-	
-	
+
+
 	public static final Logger logger = LoggerFactory.getLogger(HttpUtils.class);
 
+
+	private static CloseableHttpClient httpClient;
+
+	private HttpUtils() {
+
+	}
+
+	private static PoolingHttpClientConnectionManager cm;
+
+	private static SSLContext ctx = null;
+
+	private static SSLConnectionSocketFactory socketFactory;
+
+	private static RequestConfig requestConfig;
+
+	private static Registry<ConnectionSocketFactory> socketFactoryRegistry;
+
+	private static X509TrustManager xtm = new X509TrustManager() {
+		@Override
+		public void checkClientTrusted(X509Certificate[] chain, String authType) {
+		}
+
+		@Override
+		public void checkServerTrusted(X509Certificate[] chain, String authType) {
+		}
+
+		@Override
+		public X509Certificate[] getAcceptedIssuers() {
+			return null;
+		}
+	};
+
+	static {
+		try {
+			ctx = SSLContext.getInstance(SSLConnectionSocketFactory.TLS);
+			ctx.init(null, new TrustManager[] { xtm }, null);
+		} catch (NoSuchAlgorithmException e) {
+			logger.error("SSLContext init with NoSuchAlgorithmException", e);
+		} catch (KeyManagementException e) {
+			logger.error("SSLContext init with KeyManagementException", e);
+		}
+		socketFactory = new SSLConnectionSocketFactory(ctx, NoopHostnameVerifier.INSTANCE);
+		/** set timeout、request time、socket timeout */
+		requestConfig = RequestConfig.custom().setCookieSpec(CookieSpecs.IGNORE_COOKIES)
+				.setExpectContinueEnabled(Boolean.TRUE)
+				.setTargetPreferredAuthSchemes(Arrays.asList(AuthSchemes.NTLM, AuthSchemes.DIGEST))
+				.setProxyPreferredAuthSchemes(Arrays.asList(AuthSchemes.BASIC))
+				.setConnectTimeout(Constants.HTTP_CONNECT_TIMEOUT).setSocketTimeout(Constants.SOCKET_TIMEOUT)
+				.setConnectionRequestTimeout(Constants.HTTP_CONNECTION_REQUEST_TIMEOUT).setRedirectsEnabled(true)
+				.build();
+		socketFactoryRegistry = RegistryBuilder.<ConnectionSocketFactory>create()
+				.register("http", PlainConnectionSocketFactory.INSTANCE).register("https", socketFactory).build();
+		cm = new PoolingHttpClientConnectionManager(socketFactoryRegistry);
+		cm.setDefaultMaxPerRoute(60);
+		cm.setMaxTotal(100);
+
+	}
+
+
+	public static synchronized CloseableHttpClient getHttpClient() {
```

Comment:
public static synchronized CloseableHttpClient getHttpClient() {

This singleton mode will cause bad performance.